internal/controller/workspacetemplate_controller_test.go

```diff
@@ -274,6 +279,42 @@ var _ = Describe("WorkspaceTemplate Controller", func() {
 			Expect(requests).To(BeEmpty())
 		})
 
+		It("should return empty when workspace has incomplete template labels", func() {
+			workspace := &workspacev1alpha1.Workspace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-workspace",
+					Namespace: "default",
+					Labels: map[string]string{
+						workspaceutil.LabelWorkspaceTemplate: "my-template",
+						// Missing LabelWorkspaceTemplateNamespace
+					},
+				},
+			}
+
+			requests := reconciler.findTemplatesForWorkspace(ctx, workspace)
+			Expect(requests).To(BeEmpty(),
+				"should skip reconciliation when namespace label is missing")
+		})
+
+		It("should support cross-namespace template references", func() {
+			workspace := &workspacev1alpha1.Workspace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-workspace",
+					Namespace: "workspace-ns",
+					Labels: map[string]string{
+						workspaceutil.LabelWorkspaceTemplate:          "shared-template",
+						workspaceutil.LabelWorkspaceTemplateNamespace: "templates-ns",
+					},
+				},
+			}
+
+			requests := reconciler.findTemplatesForWorkspace(ctx, workspace)
+			Expect(requests).To(HaveLen(1))
+			Expect(requests[0].Name).To(Equal("shared-template"))
+			Expect(requests[0].Namespace).To(Equal("templates-ns"),
+				"should reconcile template in different namespace")
+		})
+
```

Comment:
fast-follow: could you test all cases of template-related label additions:
- new templateRef, templateRef.ns not empty --> add labels
- new templateRef, templateRef.ns empty --> add labels w/ defaults to ws.ns
- remove templateRef --> remove existing labels
- remove templateRef, not existing labels --> no-op